rpc/trace.go

```diff
@@ -2,16 +2,63 @@ package rpc
 
 import (
 	"context"
-	"errors"
+	"encoding/json"
 
 	"github.com/NethermindEth/juno/core/felt"
 )
 
-var ErrNotImplemented = errors.New("not implemented")
+// For a given executed transaction, return the trace of its execution, including internal calls
+func (provider *Provider) TransactionTrace(ctx context.Context, transactionHash *felt.Felt) (TxnTrace, error) {
```

Comment:
The return type should be *RPCError. 

When you return `noTraceAvailableError` in this function, what's actually returned is `noTraceAvailableError.Error()` (so we lose the code and data information).

Reply:
I'm not sure that's right.
The mock_test method for this pr returns an `error` as well and in this [line](https://github.com/NethermindEth/starknet.go/pull/328/files#diff-617ac3083426917993255a7947979919a2700fce3d4d158586d30561ee91e396R697) I return an RPCError.  When I tried in the test to compare the resulting error and the expected one it also considered the other fields, besides the error message I mean. In fact it failed becuase the data field was different

Reply:
You're right!! My bad